src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -228,7 +228,10 @@ public static IHostBuilder CreateHostBuilder(IConsole console, string[] urls, st
                     services.AddSingleton<IValidateOptions<DiagnosticPortOptions>, DiagnosticPortValidateOptions>();
 
                     services.AddSingleton<IEndpointInfoSource, FilteredEndpointInfoSource>();
+                    services.AddSingleton<IEgressOperationQueue, EgressOperationQueue>();
```

Comment:
I think we should create a "add everything that the rest server implements" extension method and make all of these an implementation detail.